File: 2022/b9012_SY.py
def pprint(stack):
    if stack:  # not empty
        print("NO")
    else:
        print("YES")


def vps(array):
    stack = []
    while array:
        ps = array.pop(0)
        if stack:  # not empty
            if (stack[-1] == ps) or (stack[-1] == ')' and ps == '('):
                stack.append(ps)
            else:
                stack.pop(-1)
        else:
            stack.append(ps)
    pprint(stack)


n = int(input())
for _ in range(n):
    vps(list(input()))

File: SCCC/b9012_SY.cpp
#include <bits/stdc++.h>
using namespace std;

string Solve() {
    string str;
    cin >> str;
    stack<char> stk;
    for (auto i : str) {
        if (i == '(') {
            stk.push(i);
        }
        else {
            if (!stk.empty()) stk.pop();
            else return "NO";
        }
    }
    if (stk.empty()) return "YES";
    else return "NO";
}


int main() {
    int n;

    cin >> n;
    for (int i = 0; i < n; i++) {
        cout << Solve() << "\n";
    }

    return 0;
}